packages/contracts-bedrock/src/L2/SoulGasToken.sol

```diff
@@ -45,6 +45,7 @@ contract SoulGasToken is ERC20Upgradeable, OwnableUpgradeable {
 
     constructor(bool isBackedByNative_) {
         IS_BACKED_BY_NATIVE = isBackedByNative_;
+        initialize("SoulGasToken", "SGT", msg.sender);
```

Comment:
We should use empty values when initialize in constructor, and we also need to adjust the code [here](https://github.com/ethstorage/optimism/blob/3834224b10c02a24a3ea874cbcb138518e11ccf2/packages/contracts-bedrock/scripts/L2Genesis.s.sol#L311-L337) accordingly.

Reply:
What changes are you referring to that need to be made in the L2Genesis? I checked the code earlier and found the following:
 - The actual implementation simply copies the code without invoking the constructor if I understand correctly.
 - The initialize() method is already called for the implementation.
 - The code follows the same practice as setL2CrossDomainMessenger.

Reply:
I mean ` SoulGasToken(impl).initialize` needs to be removed since it's now called in the constructor.

Reply:
Oops no change is required in `L2Genesis` since it used `vm.etch` to attach code for impl. The initialize call actually does nothing to the deployed code, so it's a bit confusing. However, let's just stick to the upstream convention.